### PR TITLE
test for market_manager

### PR DESCRIPTION
**Pull request by Ravleen-Solulab**
This PR contains test scripts for market_manager_abci.

### CODE REVIEW

**Review by Adamantios — CHANGES_REQUESTED** [review]
Please update the hashes and cover the following:
```
Name                                                                                     Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------------------------------------------------------------------------------

packages/valory/skills/market_manager_abci/rounds.py                                        44      4    91%   57-58, 68, 85

```

Moreover, please add the `market_manager_abci` and the merged `check_stop_trading_abci` here:
https://github.com/valory-xyz/trader/blob/b04a8265ffb9f377aa2e2319884b24a923b7ed71/tox.ini#L102-L105

**Comment by Ravleen-Solulab** [comment]
Could you please tell me specifically what did I miss?

**Comment by Adamantios** [comment]
> Could you please tell me specifically what did I miss?

Lines 57-58, 68, 85 are not covered in rounds.

File: packages/valory/skills/market_manager_abci/tests/test_rounds.py [hunk 1]
```diff
@@ -252,4 +252,9 @@ def test_market_manager_abci_app_initialization(abci_app: MarketManagerAbciApp)
 def test_synchronized_data_initialization() -> None:
     """Test the initialization and attributes of SynchronizedData."""
     data = SynchronizedData(db=AbciAppDB(setup_data={"test": ["test"]}))
+    synchronized_data = SynchronizedData(db=AbciAppDB(setup_data={"participant_to_bets_hash":["participant_to_bets_hash"]}))
+    
+    # Call the _get_deserialized method
+    deserialized_data = synchronized_data._get_deserialized
```
Comment (Adamantios) [inline packages/valory/skills/market_manager_abci/tests/test_rounds.py:258]: This is not called, it's just assigned to `deserialized_data`.

File: tox.ini [hunk 2]
```diff
@@ -99,10 +99,38 @@ commands =
...
+[commands-packages] 
+commands =
+    {[commands-base]commands}
+    pytest -rfE \
+        {env:SKILLS_PATHS}/market_manager_abci/tests \
+        --cov={env:SKILLS_PATHS}/market_manager_abci \
+        --cov-report=xml \
+        --cov-report=term \
+        --cov-report=term-missing \
+        --cov-config=.coveragerc {posargs}
```
Comment (Adamantios) [inline tox.ini:133]:
```suggestion
    pytest -rfE {env:SKILLS_PATHS}/decision_maker_abci/tests --cov={env:SKILLS_PATHS}/decision_maker_abci --cov-report=xml --cov-report=term --cov-report=term-missing --cov-config=.coveragerc {posargs}
    pytest -rfE {env:SKILLS_PATHS}/check_stop_trading_abci/tests --cov={env:SKILLS_PATHS}/check_stop_trading_abci --cov-report=xml --cov-report=term --cov-report=term-missing --cov-config=.coveragerc {posargs}
    pytest -rfE {env:SKILLS_PATHS}/market_manager_abci/tests --cov={env:SKILLS_PATHS}/market_manager_abci --cov-report=xml --cov-report=term --cov-report=term-missing --cov-config=.coveragerc {posargs}
```

File: packages/valory/skills/market_manager_abci/tests/test_rounds.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+import json
+from copy import deepcopy
```
Comment (Adamantios) [inline packages/valory/skills/market_manager_abci/tests/test_rounds.py:21]:
```suggestion
```

File: packages/valory/skills/market_manager_abci/tests/test_rounds.py [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+})
+
+
+def get_participants() -> FrozenSet[str]:
+    """Participants"""
+    return frozenset([f"agent_{i}" for i in range(MAX_PARTICIPANTS)])
+
+
+def get_payloads(
+    payload_cls: BaseTxPayload,
```
Comment (Adamantios) [inline packages/valory/skills/market_manager_abci/tests/test_rounds.py:76]:
```suggestion
    payload_cls: Type[BaseTxPayload],
```

**Review by Adamantios — APPROVED** [review]

**Pull request merged**